atemOSC/OSCReceiver.mm

```diff
@@ -553,8 +553,180 @@ - (instancetype) initWithDelegate:(AppDelegate *) delegate
 		if (IBMDSwitcherKeyFlyParameters* flyParams = [weakSelf getUSKFlyParams:key forSwitcher:s andME:me])
 			flyParams->ResetDVEFull();
 	}];
+
+	[self addEndpoint:@"/me/<me>/usk/<key>/mask/enabled" label:@"Enable Mask for USK<key>" valueType:OSCValBool handler:^void(Switcher *s, NSDictionary *d, OSCValue *v) {
+		int me = [[d objectForKey:@"<me>"] intValue];
+		int key = [[d objectForKey:@"<key>"] intValue];
+		[s keyers][me-1][key-1]->SetMasked([v boolValue]);
+	}];
+	
+	[self addEndpoint:@"/me/<me>/usk/<key>/mask/left" label:@"Set Left Mask for USK<key>" valueType:OSCValFloat handler:^void(Switcher *s, NSDictionary *d, OSCValue *v) {
+		int me = [[d objectForKey:@"<me>"] intValue];
+		int key = [[d objectForKey:@"<key>"] intValue];
+		[s keyers][me-1][key-1]->SetMaskLeft([v floatValue]);
+	}];
+	
+	[self addEndpoint:@"/me/<me>/usk/<key>/mask/right" label:@"Set Right Mask for USK<key>" valueType:OSCValFloat handler:^void(Switcher *s, NSDictionary *d, OSCValue *v) {
+		int me = [[d objectForKey:@"<me>"] intValue];
+		int key = [[d objectForKey:@"<key>"] intValue];
+		[s keyers][me-1][key-1]->SetMaskRight([v floatValue]);
+	}];
+	
+	[self addEndpoint:@"/me/<me>/usk/<key>/mask/top" label:@"Set Top Mask for USK<key>" valueType:OSCValFloat handler:^void(Switcher *s, NSDictionary *d, OSCValue *v) {
+		int me = [[d objectForKey:@"<me>"] intValue];
+		int key = [[d objectForKey:@"<key>"] intValue];
+		[s keyers][me-1][key-1]->SetMaskTop([v floatValue]);
+	}];
+	
+	[self addEndpoint:@"/me/<me>/usk/<key>/mask/bottom" label:@"Set Bottom Mask for USK<key>" valueType:OSCValFloat handler:^void(Switcher *s, NSDictionary *d, OSCValue *v) {
+		int me = [[d objectForKey:@"<me>"] intValue];
+		int key = [[d objectForKey:@"<key>"] intValue];
+		[s keyers][me-1][key-1]->SetMaskBottom([v floatValue]);
+	}];
+	
+	[self addEndpoint:@"/me/<me>/usk/<key>/mask/reset" label:@"Reset Mask for USK<key>" handler:^void(Switcher *s, NSDictionary *d, OSCValue *v) {
+		int me = [[d objectForKey:@"<me>"] intValue];
+		int key = [[d objectForKey:@"<key>"] intValue];
+		[s keyers][me-1][key-1]->ResetMask();
+	}];
+	
+	[self addEndpoint:@"/me/<me>/usk/<key>/pattern/style" label:@"Set USK<key> Pattern Style" valueType:OSCValString handler:^void(Switcher *s, NSDictionary *d, OSCValue *v) {
+		int me = [[d objectForKey:@"<me>"] intValue];
+		int key = [[d objectForKey:@"<key>"] intValue];
+		if ([[v stringValue] isEqualToString:@"bar_left2right"])
+		{
+			if (IBMDSwitcherKeyPatternParameters* patternParams = [weakSelf getUSKPatternParams:key forSwitcher:s andME:me])
```

Comment:
This section of code could be simpler by using an `NSDictionary` to store the mapping of the string value to the `bmd` constant, then indexing the dictionary to get the constant for a given OSC value.

Reply:
The result should be a dictionary definition followed by a single `SetPattern` call if the value exists in the dict, or print a useful message if not

Reply:
I'm sorry about that, but I don't know how to do… The only thing I did was copy/paste. make these changes is really out of my skills… I'm sorry

Reply:
That's fine, I can refactor after the merge